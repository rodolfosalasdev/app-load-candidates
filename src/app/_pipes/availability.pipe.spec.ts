import { AvailabilityPipe } from './availability.pipe';

describe('AvailabilityPipePipe', () => {
  let pipe: AvailabilityPipe;

  beforeEach(() => {
    pipe = new AvailabilityPipe();
  });
  
  it('create an instance', () => {
    expect(pipe).toBeTruthy();
  })
  
  it('should turn `true` into "Yes"', () => {
    expect(pipe.transform(true)).toBe('Yes');
    expect(pipe.transform(true)).toMatchSnapshot();
  });

  it('should turn `false` into "No"', () => {
    expect(pipe.transform(false)).toBe('No');
    expect(pipe.transform(false)).toMatchSnapshot();
  });

  it('should return the original value when it is not boolean', () => {
    expect(pipe.transform('other')).toBe('other');
    expect(pipe.transform(123)).toBe(123);
    expect(pipe.transform('other')).toMatchSnapshot();;
    expect(pipe.transform(123)).toMatchSnapshot();;
  });
});
