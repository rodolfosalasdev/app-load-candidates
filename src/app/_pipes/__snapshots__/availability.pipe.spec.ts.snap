// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailabilityPipePipe should return the original value when it is not boolean 1`] = `"other"`;

exports[`AvailabilityPipePipe should return the original value when it is not boolean 2`] = `123`;

exports[`AvailabilityPipePipe should turn \`false\` into "No" 1`] = `"No"`;

exports[`AvailabilityPipePipe should turn \`true\` into "Yes" 1`] = `"Yes"`;
